src/main/java/com/commercetools/sync/services/impl/BaseServiceWithKey.java

```diff
@@ -22,6 +23,8 @@
  *     io.sphere.sdk.categories.CategoryDraft}, etc..
  * @param <U> Resource (e.g. {@link io.sphere.sdk.products.Product}, {@link
  *     io.sphere.sdk.categories.Category}, etc..
+ * @param <UQ> Resource to be used on queries (e.g. {@link
```

Comment:
Maybe like this?
```suggestion
 * @param <UQ> Resource returned by the query <Q> (e.g. {@link
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/714/commits/2cfa0e15c3390988e53774876b6ea29f6af58669